### ISSUE TITLE
Temperature#apply NullPointerException on unit without value

### ISSUE THREAD

**Issue by vlovgr** [typelevel/squants#261]
```scala
@ import squants.thermal.Temperature
import squants.thermal.Temperature
@ Temperature("C")
java.lang.NullPointerException
  sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1838)
  sun.misc.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
  java.lang.Double.parseDouble(Double.java:538)
  scala.collection.immutable.StringLike.toDouble(StringLike.scala:318)
  scala.collection.immutable.StringLike.toDouble$(StringLike.scala:318)
  scala.collection.immutable.StringOps.toDouble(StringOps.scala:29)
  squants.thermal.Temperature$.apply(Temperature.scala:147)
  $sess.cmd3$.<init>(cmd3.sc:1)
  $sess.cmd3$.<clinit>(cmd3.sc:-1)
@ Temperature("F")
java.lang.NullPointerException
  sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1838)
  sun.misc.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
  java.lang.Double.parseDouble(Double.java:538)
  scala.collection.immutable.StringLike.toDouble(StringLike.scala:318)
  scala.collection.immutable.StringLike.toDouble$(StringLike.scala:318)
  scala.collection.immutable.StringOps.toDouble(StringOps.scala:29)
  squants.thermal.Temperature$.apply(Temperature.scala:146)
  $sess.cmd3$.<init>(cmd3.sc:1)
  $sess.cmd3$.<clinit>(cmd3.sc:-1)
```

(Discovered thanks to property-based tests in Ciris.)

**Comment by cquiroz** [comment]
Thanks for the report. Obviously we should produce some other type of exception in this case. However the real fix should be to get rid of those `apply` that can throw and make them return an `Option` or `Either`. That will have to wait until 2.0 though since it breaks the API

@derekmorr What do you think about adding an `fromString` method that returns `Option` and deprecate this `apply` constructor?

**Comment by derekmorr** [comment]
`Temperature.apply()` already returns a `Try`, as does every other _Dimension_.`apply()` function. I'd rather just fix the regex to handle this input.

**Comment by derekmorr** [comment]
This is fixed in PR #262. It was a one-character change to a parsing regex.

Thanks for the find, @vlovgr .
